eng/native/configureplatform.cmake

```diff
@@ -70,6 +70,11 @@ if(CLR_CMAKE_HOST_OS STREQUAL Linux)
             COMMAND bash -c "source ${LINUX_ID_FILE} && echo \$ID"
             OUTPUT_VARIABLE CLR_CMAKE_LINUX_ID
             OUTPUT_STRIP_TRAILING_WHITESPACE)
+
+        execute_process(
+            COMMAND bash -c "if strings \"${CMAKE_SYSROOT}/usr/bin/ldd\" 2>&1 | grep -q musl; then echo musl; fi"
```

Comment:
cc @janvorli, @vitek-karas, I have used same detection mechanism as we have used in: https://github.com/dotnet/runtime/blob/8e4bef2d8aaf6e710fe823d2577cd0cc1520cbdd/eng/native/init-distro-rid.sh#L163-L164 which was also ported to https://github.com/dotnet/install-scripts/blob/11b4eebe23d871c074364940d301c3eb53e7c7ec/src/dotnet-install.sh#L188

as fragile as it seems, it _just_ works.